### PR TITLE
Add Isochrone tool

### PR DESCRIPTION
**Pull request by frasercl**
Adds a new Isochrone tool for analyzing and planning traverses. Features:
- Render isochrones as raster layers showing color-ramped accumulated cost out to a set maximum
- Hover over isochrones to show least-cost paths and minimum cost from start point to hovered point
- Separated cost model interface allows easy extension and switching models on-the-fly
- Add, remove, and separately edit multiple isochrones
- Configurable color ramps, color steps, opacity, resolution, data sources

### CODE REVIEW

File: src/essence/Tools/Isochrone/IsochroneTool_Algorithm.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    [ 0,  1,  0,  9,  0],
+    [16, 15, 13, 11, 10],
+    [ 0, 14,  0, 12,  0]
+];
+
+/**
+ * Moves from a pixel to its 8 immediate neighbors.
+ * A purely distance-based model will produce an octagon with these moves.
+ */
+const queenMoves = [
```
Comment (tariqksoliman) [inline src/essence/Tools/Isochrone/IsochroneTool_Algorithm.js:141]: So she's restricting her movement to not make the king feel inferior?

**Review by tariqksoliman — APPROVED** [review]
LGTM and thanks for pulling this in!

File: src/essence/Tools/Isochrone/IsochroneTool_Algorithm.js (hunk 1 above)
Comment (frasercl) [reply src/essence/Tools/Isochrone/IsochroneTool_Algorithm.js:141]: Don't know a lot about chess haha - I believe that's a term I saw somewhere else, and I didn't think about it too much. Would probably more correctly be called `kingMoves`.

Comment (tariqksoliman) [reply src/essence/Tools/Isochrone/IsochroneTool_Algorithm.js:141]: `queen` is fine; the comment and 8 elements explain it pretty easily.

Comment (frasercl) [reply src/essence/Tools/Isochrone/IsochroneTool_Algorithm.js:141]: And those moves can compose to queen moves, so it's not as if it's completely inaccurate.

**Pull request merged**